esphome/components/esp32_ble_tracker/__init__.py

```diff
@@ -203,6 +204,7 @@ def _consume_connection_slots(config: MutableMapping) -> MutableMapping:
             cv.Optional(CONF_ON_SCAN_END): automation.validate_automation(
                 {cv.GenerateID(CONF_TRIGGER_ID): cv.declare_id(BLEEndOfScanTrigger)}
             ),
+            cv.Optional(CONF_SOFTWARE_COEXISTENCE, default=True): bool,
```

Comment:
If you use this you can remove the wifi checks below and people will get validation errors if they try to set it and wifi is not configured
```suggestion
            cv.OnlyWith(CONF_SOFTWARE_COEXISTENCE, "wifi", default=True): bool,
```

Reply:
Oh, nice. thanks !